### ISSUE TITLE
Auto code completion

### ISSUE THREAD

**Issue by prokopst** [20tab/UnrealEnginePython#384]
As someone has already pointed out in #35, there is no support for auto completion.

As far as I understand this is by design, but I (and all humans who just can't live without auto-complete) would really appropriate it. Would it be possible to at least somehow generate stub `.pyi` files for auto completion?

Something about `.pyi`
https://www.python.org/dev/peps/pep-0484/

Some keywords for potential search:
intellisense, code completion, auto complete, autocomplete, autocompletion.

**Comment by rdeioris** [comment]
hi, you can start from this example for sublime:

github.com/20tab/UnrealEnginePython/blob/master/tools/generate_sublime_text_autocomplete.py

generating a .pyi stub should not be too complex

**Comment by prokopst** [comment]
@rdeioris I've played with that and it looks like a good start, I'll prepare something and make a PR.

As a next step I would love to get arguments and return value type hints, for example:
```
def func(arg: int) -> float:
    ...
```

So IDE can hint me a lot and slap me for passing unexpected types immediately.

Are you interested in having it as part of your project or not that much? I've checked how the code is implemented and it seems that the current approach is not very type hints friendly.
